schematools/utils.py

```diff
@@ -67,6 +76,25 @@ def defs_from_url(base_url, data_type):
     return schema_lookup
 
 
+def def_from_url(base_url, data_type, dataset_name):
+    """Fetch schema definitions from a remote file for a single dataset
+    The URL could be ``https://schemas.data.amsterdam.nl/datasets/``
+    """
+    schema_lookup = {}
+    if not base_url.endswith("/"):
+        base_url = f"{base_url}/"
+
+    dataset_path = f"{dataset_name}/{dataset_name}"
+
+    with requests.Session() as connection:
+        response = connection.get(f"{base_url}{dataset_path}")
+        response.raise_for_status()
+
+        schema_lookup[dataset_name] = data_type.from_dict(response.json())
+
+    return schema_lookup
```

Comment:
Waarom geeft je hier een dict met 1 item terug ipv gewoon het item zelf?

Reply:
Wel een nuttige toevoeging, zodat we niet altijd alles hoeven op te halen en in te lezen.

Reply:
Ja goeie, heeft te maken met de vervolg logica in Permissions. Die verwacht een dict type in een functie die uiteindelijk de grants toekent. Om het logisch en expliciet te houden, geeft deze nu een schema_lookup[dataset_name] terug van type Datasetschema. In de permissies wordt daar nu rekening mee gehouden.

Reply:
Helder.